iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -356,13 +356,29 @@ public VoltageLevelExt getVoltageLevel(String id) {
 
     @Override
     public LineAdderImpl newLine() {
-        return newLine(null);
+        return newLine((String) null);
     }
 
     LineAdderImpl newLine(String subnetwork) {
         return new LineAdderImpl(this, subnetwork);
     }
 
+    @Override
+    public LineAdderImpl newLine(Line line) {
+        return newLine(null, line).setR(line.getR())
+                .setX(line.getX())
+                .setG1(line.getG1())
+                .setG2(line.getG2())
+                .setB1(line.getB1())
+                .setB2(line.getB2())
+                .setVoltageLevel1(line.getTerminal1().getVoltageLevel().getId())
+                .setVoltageLevel2(line.getTerminal2().getVoltageLevel().getId());
```

Comment:
The setters should be called in the adder's constructor (or in a utility method, but called in the constructor).
In the current state, these variables are not set when `newLine(Line)` is called from a subnetwork.

Reply:
`BranchUtil` is in the `iidm-impl` module here. I think it would be better to have this method in the `iidm-api` module, so it could be reused by custom IIDM implementations (such as powsybl-network-store).

I think you can define it as a static method directly in `LineAdder`.

Another think. It would be better to call it in `LineAdderImpl(NetworkImpl network, String subnetwork, Line copiedLine)`. It would be counterintuitive that this method initializes the limits, but not the line attributes.